modules/build/src/main/scala/scala/build/CrossSources.scala

```diff
@@ -153,6 +153,12 @@ object CrossSources {
 
     val preprocessedInputFromArgs = value(preprocessSources(inputs.flattened()))
 
+    val preprocessedWithUsingDirs =
+      preprocessedInputFromArgs.filter(p => p.options.exists(_ != BuildOptions()))
+    if (preprocessedWithUsingDirs.length > 1) logger.message(
+      "Warning: using directives detected in multiple files. It is recommended to keep them centralized in the project.scala file."
```

Comment:
can we have the current location of the `project.scala` file here?
as in, if there is a `project.scala` within the scope of the project, I think it'd be useful to point the user to its path in this warning.
(and if there is none, we could still suggest the default project root as its path)